### DISCUSSION TITLE
Module 단위가 궁금합니다.

### DISCUSSION THREAD

**Discussion by greatsk55** [nsoojin/MiniSuperApp-fastcampus#6] [아무 질문이나 환영합니다]
안녕하세요. 강의 잘 들었습니다.
특히 모듈화에 대해 인상 깊이 들었는데요. 몇가지 질문 남깁니다. 

모듈화를 View 단위로도 하나요?
올려주신 모듈화 구조도를 보면 
상위 모듈 아래의 하위 모듈들이 여러개념이 해당 될수 있겠지만

앱의 흐름으로 놓고보면 결국 View의 이동이잖아요. 
그래서 각 View들도 Module화 하는지 궁금합니다.

**Comment by nsoojin** [comment]
안녕하세요. 좋은 질문 고맙습니다.
모듈화의 단위를 여쭤보신거 같은데요, View도 모듈화를 합니다.

크게 두가지가 생각나는데,
(1) 공통 뷰들이 모여있는 UI 모듈: 보통 앱은 일관된 사용자 경험을 위해 비슷하게 생긴 UI 요소들을 사용합니다. 따라서 여러 모듈에서 공통적으로 사용하는 뷰라면 모듈화를 해서 재사용성을 높일 수 있습니다. (강의에서 만든 SuperUI 와 같은 곳에 뷰들을 넣을수 있겠죠.)
(2) 기능이 많은(코드가 많은) 뷰/화면 모듈: 많은 기능을 제공하는 뷰나 화면 단위가 있다면 개별 모듈로 만듭니다. 이 경우 public API를 제외한 객체들은 internal로 숨길수도 있고, 코드 의존성을 끊을 수 있습니다. (강의에서 만든 피쳐 모듈들에 해당)

답변이 됐을까요?
